support/testsupport/src/main/java/com/hotels/styx/support/Latch.java

```diff
@@ -0,0 +1,56 @@
+/*
+  Copyright (C) 2013-2018 Expedia Inc.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+package com.hotels.styx.support;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import static com.google.common.base.Throwables.propagate;
+
+/**
+ * Wrapper for CountDownLatch to make it more convenient to use.
+ */
+public class Latch {
+    private final CountDownLatch latch;
+
+    public Latch(int count) {
+        this.latch = new CountDownLatch(count);
+    }
+
+    public void await() {
+        try {
+            latch.await();
+        } catch (InterruptedException e) {
+            throw propagate(e);
```

Comment:
re-throwing an InterruptedException as a different Exception is not a good practice in general. I would call "Thread.interrupt()" or document it clearly when we consider it is not needed.